lib/web_ui/lib/src/engine/text/paint_service.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.12
+part of engine;
+
+class TextPaintService {
```

Comment:
Add comment:
Loop through the spans and for each box in the span:
    // 1. Paint the background rect.
    // 2. Paint the text shadows?
    // 3. Paint the text.

Reply:
This comment was more of a pseudo-implementation with the TODO. The current implementation is different from how I imagined it at the time of this comment. Right now the boxes are part of the "line" not the "span", for example. This made more sense during the implementation of gesture recognition and painting.

Added multiple comments to this file.